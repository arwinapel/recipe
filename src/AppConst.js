export const URL = 'http://localhost:4000';
export const URL_INGREDIENTS = `${URL}/ingredients`;
export const URL_RECIPES = `${URL}/recipes`;