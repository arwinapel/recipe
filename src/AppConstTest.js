const date = new Date();
const tomorrowDate = new Date(date);
tomorrowDate.setDate(tomorrowDate.getDate() + 1)
export const TOMORROW = `${tomorrowDate.getMonth() < 10 ? `0${tomorrowDate.getMonth() + 1}` : `${tomorrowDate.getMonth() + 1}`}/${tomorrowDate.getDate()}/${tomorrowDate.getFullYear()}`;
export const successIngredients = {
    data: [
        {
            "title": "Ham",
            "use-by": "2020-11-25"
        },
        {
            "title": "Cheese",
            "use-by": "2020-01-08"
        },
    ]
};
export const successRecipe = {
    data: [
        {
            "title": "Ham and Cheese Toastie",
            "ingredients": [
                "Ham",
                "Cheese",
                "Bread",
                "Butter"
            ]
        },
        {
            "title": "Salad",
            "ingredients": [
                "Lettuce",
                "Tomato",
                "Cucumber",
                "Beetroot",
                "Salad Dressing"
            ]
        },
        {
            "title": "Hotdog",
            "ingredients": [
                "Hotdog Bun",
                "Sausage",
                "Ketchup",
                "Mustard"
            ]
        }
    ]
};
export const addIngredient = jest.fn();
export const getRecipe = jest.fn();
export const ingredient = {
    title: 'Ham',
    useBy: '2019-12-12',
    addIngredient: addIngredient,
    selectedData: [],
};