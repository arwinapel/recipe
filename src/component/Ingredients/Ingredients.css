.ingredients-container {
    border: 1px solid #c7c7c7;
    border-radius: 0.3em;
    background-color: white;
    box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
    height: 80vh;
    overflow-y: auto;
}

.btn-get-recipe {
    margin-bottom: 0.5em !important;
}