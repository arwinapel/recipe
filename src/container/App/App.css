.App-container {
    background-color: var(--background-color);
    text-align: center;
    height: 100vh;
    padding: 1em;
}

.body-container {
    padding: 0.5em;
}

.btn-get-recipe {
    width: 100%;
}