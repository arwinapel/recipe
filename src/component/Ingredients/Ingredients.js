import React from "react";
import PropTypes from "prop-types";
import Ingredient from "../Ingredient/Ingredient";
import './Ingredients.css';
import List from '@material-ui/core/List';

function Ingredients({data, selectedData, addIngredient}) {
    const ingredients = data.map((el, idx) => (
        <Ingredient
            key={idx}
            selected={selectedData.includes(el.title)}
            title={el.title}
            useBy={el['use-by']}
            addIngredient={addIngredient}
        />
    ));

    return (
        <List
            dense
            disablePadding
            className={"ingredients-container"}
        >
            {ingredients}
        </List>
    )
}

Ingredients.propTypes = {
    data: PropTypes.array.isRequired,
    selectedData: PropTypes.array.isRequired,
    addIngredient: PropTypes.func.isRequired,
};

export default Ingredients