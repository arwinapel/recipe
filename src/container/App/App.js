import React, {useState} from 'react';
import './App.css';
import AppHeader from "../../component/AppHeader/AppHeader";
import axios from 'axios';
import Grid from "@material-ui/core/Grid";
import {URL_INGREDIENTS, URL_RECIPES} from "../../AppConst";
import Ingredients from "../../component/Ingredients/Ingredients";
import RecipeList from "../../component/RecipeList/RecipeList";
import Button from "@material-ui/core/Button";
import Snackbar from "@material-ui/core/Snackbar";
import {Alert} from '@material-ui/lab';

function App() {
    const [date, setDate] = useState(new Date());
    const [ingredients, setIngredients] = useState([]);
    const [recipe, setRecipe] = useState([]);
    const [error, setError] = useState(null);
    const [selectedIngredients, setSelectedIngredients] = useState([]);

    const getIngredients = async () => {
        try {
            const result = await axios.get(URL_INGREDIENTS);
            setIngredients(result.data);
        } catch (e) {
            console.log(e);
        }
    };

    const addIngredient = (title) => {
        var dataSelected = [...selectedIngredients];
        if (selectedIngredients.includes(title)) {
            dataSelected = dataSelected.filter(el => el !== title);
        } else {
            dataSelected.push(title);
        }
        setSelectedIngredients(dataSelected);
    };

    const getRecipe = async () => {
        try {
            if (selectedIngredients.length === 0) throw new Error("No ingredient selected");
            const url = `${URL_RECIPES}?.ingredients=${selectedIngredients.join(",")}`;
            const result = await axios.get(url);
            setRecipe(result.data)
        } catch (e) {
            setError(e.message)
        }
    };

    return (
        <div className="App-container">
            <AppHeader selectedDate={date} handleDateChange={setDate} getIngredients={getIngredients}/>
            <Grid container className={'body-container'} spacing={1}>
                {
                    ingredients.length > 0 &&
                    [
                        <Grid item xs={12} key={"btn-recipe"}>
                            <Button
                                className={'btn-get-recipe'}
                                data-testid={"get-recipe-btn"}
                                variant={"contained"}
                                color={"primary"}
                                disableElevation
                                onClick={getRecipe}
                            >
                                Get Recipe
                            </Button>
                        </Grid>,
                        <Grid item xs={6} key={"ingredient-container"}>
                            <Ingredients
                                data={ingredients}
                                selectedData={selectedIngredients}
                                addIngredient={addIngredient}
                                getRecipe={getRecipe}
                            />
                        </Grid>
                    ]
                }
                {
                    recipe.length > 0 &&
                    <Grid item xs={6}>
                        <RecipeList data={recipe}/>
                    </Grid>
                }
            </Grid>
            <Snackbar
                data-testid={"error-notif"}
                data-open={Boolean(error)}
                anchorOrigin={{vertical: "top", horizontal: "center"}}
                open={Boolean(error)}
                autoHideDuration={3000}
                onClose={() => setError(null)}
            >
                <Alert elevation={6} variant="filled" onClose={() => setError(null)} severity="error">
                    {error}
                </Alert>
            </Snackbar>
        </div>
    );
}

export default App;
