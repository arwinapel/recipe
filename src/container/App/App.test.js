import React from 'react';
import {cleanup, render, fireEvent, wait, waitForElement} from '@testing-library/react';
import App from './App';
import axiosMock from 'axios';
import {URL_INGREDIENTS, URL_RECIPES} from "../../AppConst";
import {addIngredient, getRecipe, successIngredients, successRecipe, TOMORROW} from "../../AppConstTest";
import Ingredients from "../../component/Ingredients/Ingredients";

jest.mock('axios');


describe("App", () => {
    afterEach(cleanup);
    it("Renders input date, value, onChange", () => {
        const {asFragment, getByTestId} = render(<App/>);
        const inputDate = getByTestId('input-date');
        expect(inputDate).toBeInTheDocument();
        expect(new Date(inputDate.value).toDateString()).toBe(new Date().toDateString());
        fireEvent.change(inputDate, {target: {value: TOMORROW}})
        expect(inputDate.value).toBe(TOMORROW)
        expect(asFragment()).toMatchSnapshot();
    });

    it("Render child", async () => {
        const {asFragment, getAllByTestId, getByTestId, rerender, queryByText} = render(<App/>);

        const getIngredientBtn = getByTestId('check-ingredient-btn');
        expect(getIngredientBtn).toBeInTheDocument();

        axiosMock.get.mockResolvedValueOnce(successIngredients);
        fireEvent.click(getIngredientBtn);
        await wait();

        const ingredients = getAllByTestId('ingredient');
        expect(ingredients.length).toBe(2);
        expect(axiosMock.get).toHaveBeenCalledTimes(1);
        expect(axiosMock.get).toHaveBeenCalledWith(URL_INGREDIENTS);

        const btnGetRecipe = getByTestId('get-recipe-btn');
        fireEvent.click(btnGetRecipe);
        expect(btnGetRecipe).toBeInTheDocument();

        const checkedIngredient = getAllByTestId('checked-ingredient');
        expect(checkedIngredient[0]).not.toHaveClass("Mui-checked");

        fireEvent.click(btnGetRecipe);
        waitForElement(() => {
            expect(getByTestId('error-notif')).toHaveAttribute('data-open', "true")
        });

        fireEvent.click(ingredients[0]);
        expect(checkedIngredient[0]).toHaveClass('Mui-checked');
        axiosMock.get.mockResolvedValueOnce(successRecipe);
        fireEvent.click(btnGetRecipe);
        await wait();
        const listRecipe = getByTestId('list-recipe');
        const titleIngredients = getAllByTestId('title-ingredient');
        expect(axiosMock.get).toHaveBeenCalledWith(`${URL_RECIPES}?.ingredients=${titleIngredients[0].textContent}`);
        expect(listRecipe).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});