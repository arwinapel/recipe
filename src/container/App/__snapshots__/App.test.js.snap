// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render child 1`] = `
<DocumentFragment>
  <div
    class="App-container"
  >
    <div
      class="appheader-container"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Choose Date
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            data-testid="input-date"
            type="text"
            value="03/20/2020"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="change date"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained btn-check-ingredient MuiButton-containedPrimary MuiButton-disableElevation"
        data-testid="check-ingredient-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Check Ingredients
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root body-container MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained btn-get-recipe MuiButton-containedPrimary MuiButton-disableElevation"
          data-testid="get-recipe-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Get Recipe
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <ul
          class="MuiList-root ingredients-container MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            data-testid="ingredient"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-496 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-497 Mui-checked MuiIconButton-colorSecondary MuiIconButton-edgeStart"
                data-testid="checked-ingredient"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="Ham"
                    class="PrivateSwitchBase-input-499"
                    data-indeterminate="false"
                    tabindex="-1"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                data-testid="title-ingredient"
              >
                Ham
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                data-testid="useby-ingredient"
              >
                2020-11-25
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
            data-testid="ingredient"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-496 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
                data-testid="checked-ingredient"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="Cheese"
                    class="PrivateSwitchBase-input-499"
                    data-indeterminate="false"
                    tabindex="-1"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
              disabled=""
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                data-testid="title-ingredient"
              >
                Cheese
              </span>
              <p
                class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                data-testid="useby-ingredient"
              >
                2020-01-08
              </p>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="recipe-list-container"
          data-testid="list-recipe"
        >
          <div
            class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="expand-panel"
          >
            <div
              aria-disabled="false"
              aria-expanded="false"
              class="MuiButtonBase-root MuiExpansionPanelSummary-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiExpansionPanelSummary-content"
              >
                Ham and Cheese Toastie
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-container MuiCollapse-hidden"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <div
                    role="region"
                  >
                    <div
                      class="MuiExpansionPanelDetails-root"
                    >
                      <ul
                        class="MuiList-root MuiList-dense MuiList-padding"
                      >
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Ham
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Cheese
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Bread
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Butter
                            </span>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="expand-panel"
          >
            <div
              aria-disabled="false"
              aria-expanded="false"
              class="MuiButtonBase-root MuiExpansionPanelSummary-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiExpansionPanelSummary-content"
              >
                Salad
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-container MuiCollapse-hidden"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <div
                    role="region"
                  >
                    <div
                      class="MuiExpansionPanelDetails-root"
                    >
                      <ul
                        class="MuiList-root MuiList-dense MuiList-padding"
                      >
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Lettuce
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Tomato
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Cucumber
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Beetroot
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Salad Dressing
                            </span>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="expand-panel"
          >
            <div
              aria-disabled="false"
              aria-expanded="false"
              class="MuiButtonBase-root MuiExpansionPanelSummary-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiExpansionPanelSummary-content"
              >
                Hotdog
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-container MuiCollapse-hidden"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <div
                    role="region"
                  >
                    <div
                      class="MuiExpansionPanelDetails-root"
                    >
                      <ul
                        class="MuiList-root MuiList-dense MuiList-padding"
                      >
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Hotdog Bun
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Sausage
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Ketchup
                            </span>
                          </div>
                        </li>
                        <li
                          class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                        >
                          <div
                            class="MuiListItemText-root MuiListItemText-dense"
                          >
                            <span
                              class="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                            >
                              Mustard
                            </span>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter"
      data-open="false"
      data-testid="error-notif"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation6"
        direction="down"
        role="alert"
        style="opacity: 0; transform: scale(0.75, 0.5625); transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 130ms cubic-bezier(0.4, 0, 0.2, 1) 65ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        />
        <div
          class="MuiAlert-action"
        >
          <button
            aria-label="Close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            title="Close"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App Renders input date, value, onChange 1`] = `
<DocumentFragment>
  <div
    class="App-container"
  >
    <div
      class="appheader-container"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Choose Date
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
            data-testid="input-date"
            type="text"
            value="03/21/2020"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              aria-label="change date"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained btn-check-ingredient MuiButton-containedPrimary MuiButton-disableElevation"
        data-testid="check-ingredient-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Check Ingredients
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root body-container MuiGrid-container MuiGrid-spacing-xs-1"
    />
  </div>
</DocumentFragment>
`;
